src/devices/Bmxx80/Bmxx80Base.cs

```diff
@@ -0,0 +1,207 @@
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="IOException"></exception>
+        protected Bmxx80Base(byte deviceId, I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+            _i2cDevice.WriteByte((byte)Bmxx80Register.CHIPID);
+
+            byte readSignature = _i2cDevice.ReadByte();
+
+            if (readSignature != deviceId)
+            {
+                throw new IOException($"Unable to find a chip with id {deviceId}");
+            }
+        }
+
+        /// <summary>
+        /// Sets the pressure sampling.
+        /// </summary>
+        /// <param name="sampling">The <see cref="Sampling"/> to set.</param>
+        public void SetPressureSampling(Sampling sampling)
+        {
+            byte status = Read8BitsFromRegister(_controlRegister);
+            status = (byte)(status & 0b1110_0011);
+            status = (byte)(status | (byte)sampling << 2);
+            _i2cDevice.Write(new[] { _controlRegister, status });
+        }
+
+        /// <summary>
+        /// Set the temperature oversampling.
+        /// </summary>
+        /// <param name="sampling">The <see cref="Sampling"/> to set.</param>
+        public void SetTemperatureSampling(Sampling sampling)
+        {
+            byte status = Read8BitsFromRegister(_controlRegister);
+            status = (byte)(status & 0b0001_1111);
+            status = (byte)(status | (byte)sampling << 5);
+            _i2cDevice.Write(new[] { _controlRegister, status });
+        }
+
+        /// <summary>
+        /// Get the current sample rate for pressure measurements
+        /// </summary>
+        /// <returns>The current pressure <see cref="Sampling"/> rate.</returns>
+        public Sampling ReadPressureSampling()
+        {
+            byte status = Read8BitsFromRegister(_controlRegister);
+            status = (byte)((status & 0b0001_1100) >> 2);
+
+            return ByteToSampling(status);
+        }
+
+        /// <summary>
+        /// Get the sample rate for temperature measurements.
+        /// </summary>
+        /// <returns>The current temperature <see cref="Sampling"/> rate.</returns>
+        public Sampling ReadTemperatureSampling()
+        {
+            byte status = Read8BitsFromRegister(_controlRegister);
+            status = (byte)((status & 0b1110_0000) >> 5);
+
+            return ByteToSampling(status);
+        }
+
+        /// <summary>
+        /// Compensates the temperature.
+        /// </summary>
+        /// <param name="adcTemperature">The temperature value read from the device.</param>
+        /// <returns>The <see cref="Temperature"/>.</returns>
+        internal Temperature CompensateTemperature(int adcTemperature)
+        {
+            //Formula from the datasheet
+            //The temperature is calculated using the compensation formula in the BMP280 datasheet
```

Comment:
NIT: can you point to the page where you got this from in the datasheet? that can save some time in the future.